apps/meteor/app/mentions/server/Mentions.ts

```diff
@@ -43,8 +43,17 @@ export class MentionsServer extends MentionsParser {
 			});
 	}
 
-	async getUsersByMentions({ msg, rid, u: sender }: Pick<IMessage, 'msg' | 'rid' | 'u'>): Promise<IMessage['mentions']> {
-		const mentions = this.getUserMentions(msg);
+	async getUsersByMentions({
+		msg,
+		rid,
+		u: sender,
+		t,
+		e2eMentions,
+	}: Pick<IMessage, 'msg' | 'rid' | 'u' | 't' | 'e2eMentions'>): Promise<IMessage['mentions']> {
+		const mentions =
+			t === 'e2e' && e2eMentions?.e2eUserMentions && e2eMentions?.e2eUserMentions.length > 0
```

Comment:
```suggestion
			t === 'e2e' && e2eMentions?.e2eUserMentions?.length > 0
```

Reply:
Not possible, we are checking the length value, if we do we get this error:
![image](https://github.com/RocketChat/Rocket.Chat/assets/20212776/55997b16-342f-4a4d-b641-256cb77e4800)